src/core/lib/security/credentials/fake/fake_credentials.c

```diff
@@ -35,21 +35,31 @@
 
 #include <string.h>
 
-#include "src/core/lib/channel/channel_args.h"
-#include "src/core/lib/iomgr/executor.h"
-
 #include <grpc/support/alloc.h>
 #include <grpc/support/log.h>
 #include <grpc/support/string_util.h>
 
+#include "src/core/ext/lb_policy/grpclb/secure_naming.h"
+#include "src/core/lib/channel/channel_args.h"
+#include "src/core/lib/iomgr/executor.h"
+
 /* -- Fake transport security credentials. -- */
 
 static grpc_security_status fake_transport_security_create_security_connector(
     grpc_exec_ctx *exec_ctx, grpc_channel_credentials *c,
     grpc_call_credentials *call_creds, const char *target,
     const grpc_channel_args *args, grpc_channel_security_connector **sc,
     grpc_channel_args **new_args) {
-  *sc = grpc_fake_channel_security_connector_create(call_creds);
+  char *expected_target = NULL;
+  if (grpc_glb_targets_info_find_in_args(args) != NULL) {
```

Comment:
It might be cleaner to not reference the target map here at all.  Instead, just look at whether GRPC_ARG_FAKE_SECURITY_EXPECTED_TARGET is present.  That way, it can be used even if cases where the target map is not specified.